server/controllers/list.js

```diff
@@ -22,28 +21,53 @@ const {
 const { updateSubdocumentFields } = require('../common/utils/helpers');
 
 const createList = (req, resp) => {
-  const { description, isListPrivate, name, userId, cohortId } = req.body;
+  const { description, isListPrivate, name, cohortId } = req.body;
+  const {
+    user: { _id: userId }
+  } = req;
 
   const list = new List({
     cohortId,
     description,
     isPrivate: isListPrivate,
+    memberIds: userId,
     name,
-    ownerIds: userId
+    ownerIds: userId,
+    viewersIds: userId
   });
 
-  list.save((err, doc) => {
-    if (err) {
-      return resp
-        .status(400)
-        .send({ message: 'List not saved. Please try again.' });
-    }
+  if (cohortId && !isListPrivate) {
+    Cohort.findOne({ _id: cohortId }).then(cohort => {
+      const { memberIds } = cohort;
+      list.viewersIds = memberIds;
 
-    resp
-      .status(201)
-      .location(`/lists/${doc._id}`)
-      .send(responseWithList(doc, userId));
-  });
+      return list.save((err, doc) => {
```

Comment:
Don't mix using callbacks and promises with queries, or you may end up with duplicate operations.
(https://mongoosejs.com/docs/queries.html)